### ISSUE TITLE
Dependency util@0.10.3 infinite loop issue on deprecate function

### ISSUE THREAD

**Issue by mkastner** [jspm/nodelibs-assert#1]
util.js runs into an ugly infinite loop, if the deprecate function (line 44) is being called. Check out line 47 of util.js where exports.deprecate calls itself.

It's already mentioned on the original author's repo, but there's no fix yet.

It ran into the problem, when running tests on the mongoose browser version which has a deprecation warning concerning its promise implementation. It took me a while to trace it back.

**Comment by guybedford** [comment]
Thanks for posting. Do you mean the function at https://github.com/jspm/nodelibs-util/blob/master/util.js#L72 then?

Can you point to the issue on the original repo?

**Comment by mkastner** [comment]
Yes, that's the one I mean. Sorry for my sloppyness. Here's the link to the original repo:

[https://github.com/defunctzombie/node-util/blob/master/util.js#L65](https://github.com/defunctzombie/node-util/blob/master/util.js#L65)

**Comment by guybedford** [comment]
Let's track movement on https://github.com/defunctzombie/node-util/issues/11 here then.
